haystack/document_stores/milvus2.py

```diff
@@ -147,22 +146,20 @@ def __init__(
             self.embedding_dim = embedding_dim
 
         self.index_file_size = index_file_size
+        self.similarity = similarity
         self.cosine = False
 
         if similarity == "dot_product":
             self.metric_type = "IP"
-            self.similarity = similarity
         elif similarity == "l2":
             self.metric_type = "L2"
-            self.similarity = similarity
         elif similarity == "cosine":
             self.metric_type = "IP"
-            self.similarity = "dot_product"
```

Comment:
@MichelBartels can you have look at this please? Is there any reason I'm missing why we were mocking the "cosine" similarity values here?

Reply:
As far as I remember, milvus2 doesn't directly support cosine similarity. https://milvus.io/docs/metric.md#Inner-product-IP There is this note:

> If you use IP to calculate embeddings similarities, you must normalize your embeddings. After normalization, the inner product equals cosine similarity.

Reply:
It's basically what Julian said: Milvus 2 only supports dot product similarity, but dot product similarity is equivalent to cosine similarity on normalized vectors. Therefore, we normalize index/query vectors before indexing/querying and set Milvus' similarity function to dot product.

However, only `self.metric_type` (IP stands for inner product i.e. dot product) and not `self.similarity` is passed to Milvus so I think what I did there was unnecessary and your change makes more sense.